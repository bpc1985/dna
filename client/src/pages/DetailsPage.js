import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import { fetchData } from "../utils";
import Loading from '../components/Loading/Loading'

import "./DetailsPage.scss";

export default function DetailsPage() {
  const { sid } = useParams();
  const [detail, setDetail] = useState();

  useEffect(() => {
    async function fetchDetailsData() {
      const data = await fetchData(`subscriptions/${sid}?_expand=dnaPackage&_expand=user`);
      setDetail(data);
    }
    fetchDetailsData();

    return () => {};
  }, []);

  if (!detail) {
    return <Loading />;
  }

  const { phone_number, address, end_user, user, dnaPackage } = detail;

  return (
    <React.Fragment>
      <h2>Subscription Details</h2>
      <div className="detailListItem">
        <div className="detailInfoLeft">
          <h1>Customer Info</h1>
          <div>Payer: {user.first_name} {user.last_name}</div>
          <div>User: {end_user}</div>
          <div>Email: {user.email}</div>
          {phone_number && <div>Phone Number: {phone_number}</div>}
          {address && <div>Address: {address}</div>}
        </div>
        <div className="detailInfoRight">
          <h1>
            <div>{dnaPackage.name}</div>
            <div>({dnaPackage.type})</div>
          </h1>
          <div>{dnaPackage.description}</div>
          <div>
            Price: â‚¬{dnaPackage.price}{" "}
            {dnaPackage.agreement !== "prepaid" && " / month"}
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}
