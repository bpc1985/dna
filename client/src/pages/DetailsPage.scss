.detailListItem {
  margin: 1rem 0;
  padding: 1rem;
  border: 1px solid #ff007c;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  .detailInfoLeft {
    width: 25%;
  }

  .detailInfoRight {
    width: 75%;
  }

  @media (max-width: 1024px) {
    .detailInfoLeft, .detailInfoRight {
      width: 100%;
    }
  }

  h1 {
    display: flex;
    justify-content: space-between;
    margin: 10px 0;
    font-size: 1.5rem;
    color: #ff007c;

    @media (max-width: 420px) {
      display: block;
    }
  }
}