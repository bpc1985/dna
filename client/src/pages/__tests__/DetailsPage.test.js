import { render, fireEvent, wait } from "@testing-library/react";
import React from "react";
import DetailsPage from "../DetailsPage";

jest.mock("react-router-dom", () => ({
  useParams: () => ({sid: '111'})
}));

jest.mock("../../utils", () => {
  const data = {
    _id: "111",
    phone_number: "111111111",
    address: "Address 1",
    end_user: "Micheal Lang",
    user: {
      first_name: "Hung",
      last_name: "Ho",
      email: "user1@mail.com"
    },
    dnaPackage: {
      name: "Package 1",
      description: "Description 1",
      type: "mobile",
      price: "99",
      agreement: "monthly"
    }
  };

  return {
    fetchSubscriptionDetailData: () => Promise.resolve(data)
  };
});

describe("DetailsPage", () => {
  it("should render component", async () => {
    const wrapper = render(<DetailsPage />);
    await wait(() => {
      expect(wrapper.container).toMatchSnapshot();
    });
  });
});
