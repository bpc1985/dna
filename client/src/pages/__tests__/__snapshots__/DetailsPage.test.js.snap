// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage should render component 1`] = `
<div>
  <h2>
    Subscription Details
  </h2>
  <div
    class="detailListItem"
  >
    <div
      class="detailInfoLeft"
    >
      <h1>
        Customer Info
      </h1>
      <div>
        Payer: 
        Hung
         
        Ho
      </div>
      <div>
        User: 
        Micheal Lang
      </div>
      <div>
        Email: 
        user1@mail.com
      </div>
      <div>
        Phone Number: 
        111111111
      </div>
      <div>
        Address: 
        Address 1
      </div>
    </div>
    <div
      class="detailInfoRight"
    >
      <h1>
        <div>
          Package 1
        </div>
        <div>
          (
          mobile
          )
        </div>
      </h1>
      <div>
        Description 1
      </div>
      <div>
        Price: â‚¬
        99
         
         / month
      </div>
    </div>
  </div>
</div>
`;
